app/single_resource/views.py

```diff
@@ -20,19 +21,32 @@ def index():
 @login_required
 def create():
     """Add a resource."""
+    descriptors = Descriptor.query.all()
+    for descriptor in descriptors:
+        # fields for option descriptors
```

Comment:
I think inline comments are better this way to distinguish from actual lengthy descriptions (e.g. of functions), esp. since these are not really full sentences